### PR TITLE
Add sticky footer

### PR DESCRIPTION
**Pull request by N47-StephanUllenboom**
Item stays stuck at the bottom of the page if it is not in the viewport.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -13,11 +13,12 @@ const getInitialConfig = el => {
 const getBindingConfig = binding => {
   const params = binding.value || {}
   const stickyTop = params.stickyTop || 0
+  const stickyBottom = params.stickyBottom !== undefined ? params.stickyBottom : false
```
Comment (ifdesign) [inline src/index.js]:
```suggestion
  const stickyBottom = params.stickyBottom
```

File: src/index.js [hunk 2]
```diff
@@ -62,7 +63,8 @@ export default {
     let childStyle = el.firstElementChild.style
 
     if (supportCSSSticky) {
-      elStyle.top = `${stickyTop}px`
+      elStyle.top = stickyBottom !== false ? 'auto' : `${stickyTop}px`
```
Comment (ifdesign) [inline src/index.js]: So stickyTop is now dependent of stickyBottom ? It should not, the element is either sticky to the top or to the bottom. This breaks the current sticky top behavior in kupo, did you check that the code is retro-compatilbe ?

File: src/index.js [hunk 3]
```diff
@@ -62,7 +63,8 @@ export default {
...
+      elStyle.bottom = stickyBottom !== false ? `${stickyTop}px` : ''
```
Comment (ifdesign) [inline src/index.js]: Why is sticky-bottom applying sticky-top value ?

File: src/index.js [hunk 4]
```diff
@@ -104,8 +107,14 @@ export default {
     }
 
     instance.listenAction = throttle(() => {
+      const rect = el.getBoundingClientRect()
       const offsetTop = el.getBoundingClientRect().top
-      if (offsetTop <= stickyTop) return sticky()
+
+      if (stickyBottom !== false) {
```
Comment (ifdesign) [inline src/index.js]:
```suggestion
      if (stickyBottom) {
```

File: src/index.js (hunk 3 above)
Comment (N47-StephanUllenboom) [reply src/index.js]: thats not correct right

File: src/index.js (hunk 4 above)
Comment (N47-StephanUllenboom) [reply src/index.js]: This is not correct - because 0 is also now not triggered.

File: src/index.js (hunk 2 above)
Comment (N47-StephanUllenboom) [reply src/index.js]: I cant test the kupo project, sorry. But I test the different sticky mods. You need auto is correct for the sticky bottom behavior. I dont understand why this line will broke the top behavior in kupo?

File: src/index.js (hunk 1 above)
Comment (N47-StephanUllenboom) [reply src/index.js]: okay, that will also work. But we have to change then all L#66f, 86 and 113

**Comment by N47-StephanUllenboom** [comment]
> Please replace myVar !== undefined ? 'a' : 'b' and myVar !== false ? 'a' : 'b' with myVar ? 'a' : 'b' and myVar !== undefined ? myVar : anotherVar with myVar || anotherVar.

That  is not correct, if `myVar = 0` then you get in each case the wrong value (anotherVar)!

> Verify that the changes don't break the current implementation where it is used.
I will check this again

> Also check that without the stickyBottom parameter, everything works correctly
I will check this again

> Don't make stickyBottom and stickTop dependent from each other
We need in some  cases the dependenies

File: src/index.js (hunk 2 above)
Comment (N47-StephanUllenboom) [reply src/index.js]: And no stickyTop is not dependent but you can't have both. stickyTop is the default one

**Pull request merged**